stingray/gti.py

```diff
@@ -1020,6 +1079,51 @@ def time_intervals_from_gtis(gtis, chunk_length, fraction_step=1,
     return spectrum_start_times, spectrum_start_times + chunk_length
 
 
+def get_segment_bin_start(startbin, stopbin, nbin, fraction_step=1):
+    """Get start of intervals of equal length, given bin number range
+
+    Parameters
+    ----------
+    startbin : int
+        Starting bin of the interval
+
+    stopbin : int
+        Last bin of the interval
+
+    nbin : int
+        number of bins in each interval
+
+    Other Parameters
+    ----------------
+    fraction_step : float
+        If the step is not a full ``chunk_length`` but less (e.g. a moving window),
+        this indicates the ratio between step step and ``chunk_length`` (e.g.
+        ``0.5`` means that the window shifts of half ``chunk_length``)
+
+    Returns
+    -------
+    spectrum_start_bins : array-like
+        List of starting bins in the original time array to use in spectral
+        calculations.
+
+    Examples
+    --------
+    >>> st = get_segment_bin_start(0, 10000, 10000)
+    >>> st[-1]
+    0
+    >>> st = get_segment_bin_start(0, 5, 2)
+    >>> st[-1]
+    2
+    >>> st = get_segment_bin_start(0, 6, 2)
+    >>> st[-1]
+    4
+    """
+    st = np.arange(startbin, stopbin, int(nbin * fraction_step), dtype=int)
+    if st[-1] + nbin > stopbin:
+        return st[:-1]
+    return st
+
+
 def bin_intervals_from_gtis(gtis, chunk_length, time, dt=None, fraction_step=1,
```

Comment:
I feel like there's a bunch of functions here that all seem to have vaguely similar but subtly different purposes, with names and descriptions that don't necessarily make it very easy (to me) to figure out how and why they're different. Like, I'm having trouble understanding how this function is different compared to the previous one?

Reply:
This one gives the start and stop of bin intervals in the whole observation, from all GTIs, the previous one gives the starting bins of intervals in a range of bins deemed continuous (e.g. inside a GTI, with a uniformly sampled light curve)

Reply:
Okay, I'm not sure that helped my understanding significantly, to be honest. :D 
Partly, I'm really confused around terms like "bin intervals", "time intervals", "chunk_length", "segment_length", "nbins" and how they all fit together? Are "chunk_length" and "segment_length" the same? If so, I think it would be helpful to unify the language. 

So maybe this could be called `generate_segment_idx_from_gti`? And I don't think it generates the start/stop times, but finds the start and stop indices of equally long segments that are compatible with GTIs?